RabbitMQ.xs

```diff
@@ -530,62 +515,155 @@ int internal_recv(HV *RETVAL, amqp_connection_state_t conn, int piggyback, int t
 
           // Handle tables (hashes when translated to Perl)
           case AMQP_FIELD_KIND_TABLE:
-            hv_store(
-              headers,
+            hv_store( headers,
               header_entry->key.bytes, header_entry->key.len,
               mq_table_to_hashref( &header_entry->value.value.table ),
               0
             );
             break;
 
           default:
-            Perl_croak(aTHX_ "Unsupported AMQP kind '%c' detected.", (unsigned char)header_entry->value.kind);
+            ret.reply_type = AMQP_RESPONSE_LIBRARY_EXCEPTION;
+            ret.library_error = AMQP_STATUS_UNKNOWN_TYPE;
+            goto error_out2;
```

Comment:
I don't care for the `goto` pattern you've got here, it seems like it would be just as easy to return a macro for this. That said, this is a stylistic thing that several projects written in C disagree with, so I am okay with them coming in.

Could we name the labels better though?